internal/filematcher/match.go

```diff
@@ -0,0 +1,37 @@
+package filematcher
+
+import (
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+)
+
+// Match returns list of files and dirs matching a glob pattern
+func Match(pattern string) ([]string, error) {
+	if IsRemoteFile(pattern) {
+		return []string{pattern}, nil
+	}
+	files, err := filepath.Glob(pattern)
+	if err != nil {
+		return nil, err
+	}
+	// files is nil when pattern does not match any files
+	if files == nil {
+		return nil, fmt.Errorf("%s: %w", pattern, fs.ErrNotExist)
+	}
+	var envFiles []string
+	for _, f := range files {
+		abs, err := filepath.Abs(f)
+		if err != nil {
+			return nil, err
+		}
+		envFiles = append(envFiles, abs)
+	}
+	return envFiles, nil
```

Comment:
needs a sort because `Glob` does not guarantee this and we always need a predictable order

Reply:
also update the function comment to reflect this